pywbem/_logging.py

```diff
@@ -14,404 +14,341 @@
 
 #: Name of logger for HTTP requests and responses between the pywbem client and
 #: WBEM servers.
-LOG_HTTP_NAME = 'pywbem.http'
+LOGGER_HTTP_NAME = 'pywbem.http'
 
-#: List of the logger names that :class:`~pywbem._logging.PywbemLoggers`
-#: recognizes.
-LOG_COMPONENTS = ['ops', 'http', 'all']
+#: List of the logger names that the logging configuration helper functions
+#: recognize. This allows setting up loggers individually or all to the
+#: same destination.
+LOGGER_SIMPLE_NAMES = ['api', 'http', 'all']
 
-#: List of log destinations that :class:`~pywbem._logging.PywbemLoggers`
-#: recognizes.
+#: List of log destinations that the logging configuration helper functions
+#: recognize.
 LOG_DESTINATIONS = ['file', 'stderr', 'none']
 
-#: List of the log detail strings that :class:`~pywbem._logging.PywbemLoggers`
-#: recognizes.
-LOG_DETAIL_LEVELS = ['all', 'min']
+#: List of the log detail strings that the logging configuration helper
+#: functions recognizes when formatting log records.
+#:
+#:  * `all` - All of the data available is output. Generally this is
+#:    the repr of the objects in the request or response
+#:
+#:  * `summary` - Only summary information is output. For api responses this
+#:    is primarily the count and type of objects returned. For http logs
+#:    the HTTP header information is output.
+#:
+#:  * `paths` - For api responses that include CIMInstanceName or CIMClassName
+#:    objects(paths components in the objects) only the string representation
+#:    of the path components is output.  Otherwise the complete object
+#:    representations are logged.
+#:
+#:  * :class:`py:int` - All of the available data is formatted up to the size
+#:    defined. by the integer value for each log record.  This option applies
+#:    to all of the logs generated.
+LOG_DETAIL_LEVELS = ['all', 'paths', 'summary']
 
 #: Default log detail level string if none is supplied with a call to the
 #: :class:`pywbem._logging.PywbemLoggers` methods that configure pywbem named
 #: loggers
-DEFAULT_LOG_DETAIL_LEVEL = 'min'
-
-#: Default maximum log entry size if none is supplied when creating
-#: :class:`~pywbem.LogOperationRecorder` objects.
-DEFAULT_MAX_LOG_ENTRY_SIZE = 1000
+DEFAULT_LOG_DETAIL_LEVEL = 'all'
 
 #: Default log destination if none is defined when named loggers are
 #: configured. 'none' means that there is no logging.
 DEFAULT_LOG_DESTINATION = 'none'
 
 
-class MetaPywbemLoggers(type):
-    """
-    **Experimental:** *New in pywbem 0.11 as experimental.*
-
-    This metaclass allows the definition of __str__ and __repr__ at the
-    class level in the subclass.
+def configure_loggers_from_string(log_config_str, log_filename=None,
+                                  connection=None):
+    # pylint: disable=line-too-long
     """
-    loggers = {}
-
-    def __str__(cls):
-        return 'PywbemLoggers(loggers={s.loggers!r})'.format(s=cls)
+    Configure the pywbem loggers defined by the input string in the syntax
```

Comment:
There are multiple input strings. While it is rather clear which one is meant, I still suggest to name the parameter.